### ISSUE TITLE
The Proposal to improve NgRestModel

### ISSUE THREAD

**Issue by cheerymagicsoftware** [luyadev/luya#1776]
The Proposal to improve. 
When we define fields in public property $i18n of \luya\admin\ngrest\base\NgRestModel 
and then save/update them from frontend (or another plases), they are stored in raw form without "json" wrapper. Example:
We have 2 Languages in admin [en, de]

	class Article extends NgRestModel
	{
		...
		
		public $i18n = ['name', 'sku'];
		
		....
	}

Frontend Controller:

	class DefaultController extends Controller
	{
		public function actionParse()
		{
			$article = Article::find()->where(['id' => '1'])->one();
			$article->name = 'Jeans';
			$article->sku = 482452355125;
			$article->save();
		}
               ...
	}

tbl_article
name 	| sku
Jeans	| 482452355125

In admin area table these values is not visible!

Proposal: add to NgRestModel method beforeSave (it's for string values :)):

	abstract class NgRestModel extends ActiveRecord implements GenericSearchInterface, NgRestModelInterface
	{
		...
		public function beforeSave($insert)
		{
			foreach($this->attributes as $attr => $value){
				if($this->isI18n($attr) && !\luya\helpers\Json::isJson($value)) {
					$getLang = \Yii::$app->adminLanguage->getLanguages();
					$langs = \yii\helpers\ArrayHelper::getColumn($getLang,'short_code');
					$multiValue = [];
					foreach($langs as $lang => $val) {
						$multiValue[$lang] = $value;
					}
					$this->{$attr} = \luya\admin\helpers\I18n::encode($multiValue);
				}
			}

			return parent::beforeSave($insert);
		}
		...		
	}
	
After that 
tbl_article
name 								| sku
{"en":"Jeans","de":"Jeans"}			| {"en":"482452355125","de":"482452355125"}
Thanks.

**Comment by nadar** [comment]
hi @cheerymagicsoftware thanks for making LUYA better, and thinking about new ways to improve LUYA and its core concept NgRest.

Currently i am not sure why it makes sense to store the same value for different language? Why not then disabled the i18n field for those fields?

But yes maybe we should think about your solution.

**Comment by nadar** [comment]
@cheerymagicsoftware i am still thinking about your input. It should have the possibility to set language for all current languages like

```php
$article->sku = ['de' => 482452355125, 'en' => 1234567]
```

But saving the same value into both languages does not make sense to me, or could you please explain the use case? Because we should be very careful with "magical behaviors".

**Comment by cheerymagicsoftware** [comment]
Indeed, I don't need i18n in fields like sku or birthday or TIN... but always i18n is needed in title, name, description etc.

**Comment by nadar** [comment]
@cheerymagicsoftware But maybe we could say:

1. If its an i18n field
2. And the input is an array
3. The array will be encoded to the json

Like:

```
$article->title = ['de' => 'FooBar', 'en' => 'BarFoo'];
```

This would not force you to do so, but if its an array it would work.

What do you think?
